framework/include/materials/MaterialData.h

```diff
@@ -161,6 +176,12 @@ class MaterialData
   /// Status of storage swapping (calling swap sets this to true; swapBack sets it to false)
   bool _swapped;
 
+  /// Properties having dot properties
+  std::vector<const MaterialProperty<Real> *> _current_props;
+  std::vector<const MaterialProperty<Real> *> _old_props;
+  std::vector<const MaterialProperty<Real> *> _older_props;
```

Comment:
So we have to basically add three data members that conceptually mimic the members 10 lines earlier?

Reply:
And `_props_old` vs. `_old_props` ?! The new names offer no indication of what sets these members apart from the properties above.

Reply:
These three members only contains properties in REAL that are to be used for evaluating dot properties. They are used in time integrators. We can have a vector of bools in size of `_props.size()` to indicate if the properties are going to be used for dot evaluation. We will have to cast the property into real later.

Reply:
That sounds better